### PR TITLE
Add backend test modules

### PR DESCRIPTION
**Pull request by sonowz**
Currently left for CI pass testing

Changes:
- Added exception handling
- Removed unnecessary dependency in `TreeLoader`
- Refactored `TreeNode` constructor

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ install:
 
 # command to run tests
 script:
-  - python backend/manage.py test
+  - "cd backend && python manage.py test --keepdb && cd .."
```
Comment (sonowz) [inline .travis.yml:21]: Using `python backend/manage.py test` does not detect any tests.
Maybe `BASE_DIR` problem?

File: backend/core/rule/tree.py [hunk 2]
```diff
@@ -171,10 +175,11 @@ def tree_into_dict(self, hide=False):
     def add_children(self, obj):
         self.children.append(obj)
 
-    def eval_children(self):
+    def eval_children(self, taken_list):
```
Comment (sonowz) [inline backend/core/rule/tree.py:178]: `sugang_list` is now given when evaluating tree.

**Comment by sonowz** [comment]
`python 2.7` and `pypy` doesn't support loading protocol of `test_sample/sugang_list.pickle`.
We should decide whether delete those environments, or find other way.

**Comment by dj-shin** [comment]
Protocol for pickle differs between 2 and 3 (I'm not sure of pypy). We can either find a way to make it compatible, or we can simply ignore other than python3. As I have mentioned in #35, I prefer to make it available for multiple environments, but not a must when it requires too much effort.

I guess it's better to remove other versions, for our primary goal is not compatibility.

**Comment by sonowz** [comment]
I think keeping `python 2` compatible is nothing but using 2 rather than 3.
So I'm removing `python 2` and `pypy`.

**Review by dj-shin — APPROVED** [review]

**Pull request merged**